petl/io/json.py

```diff
@@ -175,9 +205,43 @@ def __iter__(self):
 
 class DictsGeneratorView(DictsView):
 
+    def __init__(self, dicts, header=None, sample=1000, missing=None):
+        super(DictsGeneratorView, self).__init__(dicts, header, sample, missing)
+        self._filecache = None
+
     def __iter__(self):
-        self.dicts, dicts = itertools.tee(self.dicts)
-        return iterdicts(dicts, self._header, self.sample, self.missing)
+        if not self._header:
+            self._determine_header()
+        yield self._header
+
+        if not self._filecache:
+            self._filecache = NamedTemporaryFile(delete=False, mode='wb')
+            it = iter(self.dicts)
+            for o in it:
+                row = tuple(o[f] if f in o else self.missing for f in self._header)
```

Comment:
```suggestion
                row = tuple(o.get(f, self.missing) for f in self._header)
```

Reply:
Commited in the branch that has superseded this one: https://github.com/petl-developers/petl/pull/626/commits/662b7857da329e23f8d7ef007b36aa3181fd7177